orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java

```diff
@@ -82,9 +74,10 @@ public OrchestrationClient(@Nonnull final AiCoreDeployment deployment) {
    * @throws OrchestrationClientException if the request fails
    */
   @Nonnull
-  public String chatCompletion(@Nonnull final String userPrompt)
+  public String chatCompletion(
+      @Nonnull final String userPrompt, @Nonnull final OrchestrationConfig config)
```

Comment:
I would have `OrchestrationClient.withConfig(config).chatCompletion("prompt")`

Reply:
But then what would be the result of `OrchestrationClient.withConfig(config)` alone? A `PromptBuilder` or something like that?

Reply:
The config is stored on the client and `withConfig()` returns `this`